File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/inventory/items/ObjectMovementMessage.scala





















// Generated on 05/03/2013 10:34:57
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait ObjectMovementMessage extends MessageDeserializer {
  val messageId = 3010

  protected trait Message extends BaseMessage with Serializable {
    val messageId = ObjectMovementMessage.messageId

    val objectUID: Int
    val position: UByte

    def serialize(buf: Buffer) {
      buf.writeInt(objectUID)
      buf.writeUByte(position)
    }

  }

  protected def create(objectUID: Int, position: UByte): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val objectUID = buf.readInt()
    val position = buf.readUByte()

    create(objectUID, position)
  }

}

object ObjectMovementMessage extends ObjectMovementMessage {
  case class MessageImpl(objectUID: Int, position: UByte) extends Message
  type Target = MessageImpl

  override def create(objectUID: Int, position: UByte) = MessageImpl(objectUID, position)

  def apply(objectUID: Int, position: UByte) = create(objectUID, position)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/ObjectMovementMessage.java


// Generated on 05/08/2013 19:37:56
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class ObjectMovementMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 3010;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public int objectUID;
    public short position;
    
    public ObjectMovementMessage() { }
    
    public ObjectMovementMessage(int objectUID, short position) {
        this.objectUID = objectUID;
        this.position = position;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeInt(objectUID);
        buf.writeUByte(position);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        objectUID = buf.readInt();
        if (objectUID < 0)
            throw new RuntimeException("Forbidden value on objectUID = " + objectUID + ", it doesn't respect the following condition : objectUID < 0");
        position = buf.readUByte();
        if (position < 0 || position > 255)
            throw new RuntimeException("Forbidden value on position = " + position + ", it doesn't respect the following condition : position < 0 || position > 255");
    }
    
}
